R/occumb.R

```diff
@@ -768,7 +768,19 @@ set_design_matrix <- function(formula, list_cov, omit_intercept = FALSE) {
         out <- tryCatch({
           stats::model.matrix(formula, data = list_cov)
         }, warning = function(w) {
-          stop(conditionMessage(w), "\n")
+          atr <- attr(terms(formula), "variable")
+          stats_list <- lapply(ls("package:stats"), as.name)
```

Comment:
`stats_list` contains many functions that shouldn't cause an issue. It will be essential that the object used for this purpose contains a necessary and sufficient list of functions that you want to capture as problematic. It may have only `poly` right now, but if you find something else you need to add later, you can add it manually.

Reply:
Sounds like it might be too complicate to capture only the functions which throws error/warning when character variables are used so I agree on adding specific cases like `poly` manually when needed.